apollo-normalized-cache-sqlite/src/iosMain/kotlin/com/apollographql/apollo/cache/normalized/sql/internal/DatabaseRequestExecutor.kt

```diff
@@ -0,0 +1,83 @@
+package com.apollographql.apollo.cache.normalized.sql.internal
+
+import com.apollographql.apollo.cache.normalized.sql.ApolloDatabase
+import kotlinx.cinterop.COpaquePointer
+import kotlinx.cinterop.StableRef
+import kotlinx.cinterop.asStableRef
+import kotlinx.cinterop.autoreleasepool
+import kotlinx.cinterop.staticCFunction
+import platform.Foundation.NSThread
+import platform.darwin.dispatch_async_f
+import platform.darwin.dispatch_get_main_queue
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.suspendCoroutine
+import kotlin.native.concurrent.TransferMode
+import kotlin.native.concurrent.Worker
+import kotlin.native.concurrent.freeze
+
+/**
+ * Performs database operation on background thread via [Worker] API.
+ * This executor requires caller thread to be main.
+ * After database operation execution it resumes on main thread.
+ */
+internal actual class DatabaseRequestExecutor actual constructor(
+    private val database: ApolloDatabase
+) {
+  private val worker: Worker = Worker.start()
```

Comment:
Any pro/cons of using a custom queue using `dispatch_queue_create()` instead of a `Watcher` and use the same mecanism you used below to dispatch on the main thread ? My current understanding is that the Worker API is also available on other platforms like linux/windows but since we're in Apple world here, dispatch queue will always be available and might be a more platform idiomatic way to do?

Reply:
Yeah we can do that, I think Worker is thread confined and execution of all tasks is serialized. Also if we use CGD means if it spins a new thread we need to initialize Kotlin Runtime on it. Not sure how bad it affects perf

Reply:
Good point. Since it's working already, let's keep it like this 👍